rholang/src/main/scala/AbstractInterpreter/RhoInterface.scala

```diff
@@ -0,0 +1,453 @@
+      )
+    )
+  )
+
+  // new x in { x!(0) | for(z <- x){*z} }
+  val reducible_8 = List(
+    New[Channel](
+      Var("x"),
+      Par(
+        Output(
+          Var("x"),
+          Zero()
+        ),
+        Input(
+          Var("z"),
+          Var("x"),
+          Drop(Var("z"))
+        )
+      )
+    )
+  )
+
+  // new x in { x!(0) | for(z <- x){ *z } }
+  val reducible_9 = List(
+    New(
+      Var("x"),
+      Par(
+        Output(
+          Var("x"),
+          Zero()
+        ),
+        Input(
+          Var("z"),
+          Var("x"),
+          Drop(Var("z"))
+        ),
+        Input(
+          Var("u"),
+          Var("x"),
+          Drop(Var("u"))
+        )
+      )
+    )
+  )
+
+  // new x in { new y in { x!(*y) | for(u <- y){ u!0 }} | for( z <- x ){ z!(0) } }
+  val reducible_10 = List(
+    New(
+      Var("x"),
+      Par(
+        New(
+          Var("y"),
+          Par(
+            Output(Var("x"), Drop(Var("y"))),
+            Input(Var("u"), Var("y"), Output(Var("u"), Zero()))
+          )
+        ),
+        Input(Var("z"), Var("x"), Output(Var("z"), Zero()))
+      )
+    )
+  )
+
+  // new x in { @(0|0)!*x | for(z <- @0){ new y in { z!(y!(0)) | for(v <- y){ *v } | for(q <- z){ *q }}}}
+
+  val reducible_11 = List(
+    New[Channel](
+     Var("x"),
+      Par(
+        Output(Quote(Par(Zero(),Zero())),Drop(Var("x"))),
+        Input(
+          Var("z"),
+          Quote(Par(Zero(),Zero())),
+          New(
+            Var("y"),
+            Par(
+              Output(Var("z"),Output(Var("y"),Zero())),
+              Input(Var("v"),Var("y"),Drop(Var("v"))),
+              Input(Var("q"),Var("z"),Drop(Var("q")))
+            )
+          )
+        )
+      )
+    )
+  )
+
+  val result = RhoInterface.reduceM.reduce(MachineState(HashMap.empty[Channel, ChannelQueue], reducible_11))
+
+  println(result.length)
+  println(result.map{triple => triple._3.mkString("\n")}.mkString("\n" + "Terminated" + "\n" + "\n"))
+}
+
+//A channel may be a quoted process, or a variable.
+sealed trait Channel
+
+  case class Quote(unquote: Proc[Channel]) extends Channel {
+    override def toString: String = "@(" + unquote + ")"
+  }
+
+  case class Var(id: String) extends Channel {
+    override def toString: String = id
+  }
+
+//A ChannelQueue may be an empty queue, a list of readers, or a list of writers.
+// It will never be both a list of readers and a list of writers
+sealed trait ChannelQueue
+
+  case class ReaderQueue(x:Reader, xs: List[Reader]) extends ChannelQueue {
+    override def toString: String = (x :: xs).map{_.toString}.mkString("[", "][", "]")
+  }
+
+  case class WriterQueue(x: Writer, xs: List[Writer]) extends ChannelQueue {
+    override def toString: String = (x :: xs).map{_.toString}.mkString("[", "][", "]")
+  }
+
+  case class EmptyQueue() extends ChannelQueue {
+    override def toString: String = "[]"
+  }
+
+//A reader is a abstraction providing a bound name "z", and a continuation
```

Comment:
All these outer multi-line comments should be in this format:
```
/**
  * A reader is [...]
  */
````
If it is just one line then: `/** A reader is [...] */`